src/components/ArticleList.jsx

```diff
@@ -2,35 +2,42 @@ import { useEffect, useState } from "react";
 import { ArticleSummary } from "./ArticleSummary";
 import { apiClient } from "../client";
 import "../styles/ArticleList.css";
+import { Pagination } from "./Pagination";
 
 export function ArticleList() {
   const [articles, setArticles] = useState([]);
   const [totalCount, setTotalCount] = useState(0);
   const [isArticlesLoading, setIsArticlesLoading] = useState(true);
+  const [listPagination, setListPagination] = useState({ limit: 10, p: 1 });
 
   useEffect(() => {
     setIsArticlesLoading(true);
     apiClient
-      .get("articles")
+      .get("articles", { params: { ...listPagination } })
       .then(({ data: { articles, total_count } }) => {
         setArticles(articles);
         setTotalCount(total_count);
-        console.log(articles[0]);
       })
       .catch(console.log)
       .finally(setIsArticlesLoading(false));
```

Comment:
Move this out of a .finally and into the .then, after you set the articles. It should work then for the loading pattern! I'm not 100% sure on the order of a .finally with .then and catch - but it doesn't seem to play ball with the asynchronicity!

Reply:
Thanks!